client/src/api/v1/index.ts

```diff
@@ -7,7 +7,7 @@ export default class ServerApi {
     this.api = axios.create({ baseURL: `api/v1` });
   }
 
-  async post(path: string, form: any) {
+  protected async post<Type>(path: string, form: Type) {
```

Comment:
I would make the return type a generic as well. E.g. sign-in IIRC returns some user info.

https://www.oreilly.com/library/view/learning-typescript-2x/9781788391474/b170884f-368a-417e-9ba4-b5e86207b8dd.xhtml